Assets/Resources/Shaders/TransparentLinePortalShader.shader

```diff
@@ -35,68 +34,68 @@
             #pragma multi_compile _ PIXELSNAP_ON
             #pragma multi_compile _ ETC1_EXTERNAL_ALPHA
             #include "UnityCG.cginc"
-           
+
             struct appdata_t
             {
                 float4 vertex   : POSITION;
                 float4 color    : COLOR;
                 float2 texcoord : TEXCOORD0;
                 UNITY_VERTEX_INPUT_INSTANCE_ID
             };
- 
+
             struct v2f
             {
                 float4 vertex    : SV_POSITION;
-				float3 worldPos  : TEXCOORD1;
+                float3 worldPos  : TEXCOORD1;
                 fixed4 color     : COLOR;
                 float2 texcoord  : TEXCOORD0;
                 UNITY_VERTEX_OUTPUT_STEREO
             };
-           
+
             fixed4 _Color;
- 
+
             v2f vert(appdata_t IN)
             {
                 v2f OUT;
                 UNITY_SETUP_INSTANCE_ID(IN);
                 UNITY_INITIALIZE_VERTEX_OUTPUT_STEREO(OUT);
                 OUT.vertex = UnityObjectToClipPos(IN.vertex);
-				OUT.worldPos = mul(unity_ObjectToWorld, float4(IN.vertex.x, IN.vertex.y, IN.vertex.z, 1.0)).xyz;
+                OUT.worldPos = mul(unity_ObjectToWorld, IN.vertex);
                 OUT.texcoord = IN.texcoord;
                 OUT.color = IN.color * _Color;
                 #ifdef PIXELSNAP_ON
                 OUT.vertex = UnityPixelSnap (OUT.vertex);
                 #endif
- 
+
                 return OUT;
             }
- 
+
             sampler2D _MainTex;
             sampler2D _AlphaTex;
-			float2 _PortalMin;
-			float2 _PortalMax;
- 
+            float4 _Portal;
+
             fixed4 SampleSpriteTexture (float2 uv)
             {
                 fixed4 color = tex2D (_MainTex, uv);
- 
+
 #if ETC1_EXTERNAL_ALPHA
                 // get the color from an external texture (usecase: Alpha support for ETC1 on android)
                 color.a = tex2D (_AlphaTex, uv).r;
 #endif //ETC1_EXTERNAL_ALPHA
- 
+
                 return color;
             }
- 
+
             fixed4 frag(v2f IN) : SV_Target
             {
+                // Discard coordinates if transparent or outside portal
+                // Note: We use a 2D portal that spans over Unity's XZ plane: (x_min, z_min, x_max, z_max)
+                if (IN.worldPos.x < _Portal.x || IN.worldPos.z < _Portal.y ||
+                    IN.worldPos.x > _Portal.z || IN.worldPos.z > _Portal.w)
+                {
+                    discard;
+                }
                 fixed4 c = SampleSpriteTexture (IN.texcoord) * IN.color;
-				if (IN.worldPos.x < _PortalMin.x || IN.worldPos.z < _PortalMin.y ||
-					IN.worldPos.x > _PortalMax.x || IN.worldPos.z > _PortalMax.y
-				)
-				{
-					c.a = 0.0f;
-				}
                 c.rgb *= c.a;
```

Comment:
Do we still need this line, now that we no longer use the alpha value to hide fragments outside the portal?

Reply:
I thought it was part of the alpha blending. Transparency can be applied via texture.

If you know more, or want to spend some time investigating, please feel free :)

Reply:
> I thought it was part of the alpha blending. Transparency can be applied via texture.

That makes sense, I forgot that the texture can apply transparency too, so let's keep it as-is.